spring-data-jdbc/src/main/java/org/springframework/data/jdbc/core/convert/MappingJdbcConverter.java

```diff
@@ -185,33 +173,48 @@ private Class<?> doGetColumnType(RelationalPersistentProperty property) {
 	}
 
 	@Override
-	@Nullable
-	public Object readValue(@Nullable Object value, TypeInformation<?> type) {
-
-		if (value == null) {
-			return value;
-		}
+	protected Object readTechnologyType(Object value) {
 
 		if (value instanceof Array array) {
 			try {
-				return super.readValue(array.getArray(), type);
-			} catch (SQLException | ConverterNotFoundException e) {
+				return array.getArray();
```

Comment:
What is the typical case for a `SQLException`? Leniently falling back to standard conversions seems to hide a potential issue. We would rather want to propagate the exception, as for other exceptions (mostly runtime exceptions) we don't fall back to some exception handling and back off but the exception is being thrown to the caller.